### PR TITLE
Guests hotplugging split

### PR DESCRIPTION
**Pull request by tbaev**
Separate module hotplugging.md in smaller tests for easier debugging.   

- Related ticket: https://progress.opensuse.org/issues/96230
- Needles: N/A
- Verification run: [XEN](http://openqa.qam.suse.cz/tests/34734#dependencies)  [KVM](http://openqa.qam.suse.cz/tests/34744#dependencies)

### CODE REVIEW

**Comment by pdostal** [comment]
I would keep the new subroutines from `lib/virt_autotest/utils.pm` in corresponding `hotplugging_*` test module as they are specific to it. Otherwise this looks good to me!

**Comment by tbaev** [comment]
The corresponding subroutines in `lib/virt_autotest/utils.pm` are used across multiple `hotplugging_*` test modules. 
If I put them in the corresponding 'hotplugging_*' I will get an error from CI/CD about code repetition.

Maybe I can create `tests/virtualization/universal/hotplugging_utils.pm` and put the subroutines there?

**Comment by pdostal** [comment]
Aha, I see now - you call `reset_guest()` in basically every `hotplugging_*` test module.

Can you split `reset_guest()` into `reset_guest_nics()`, `reset_guest_disks()`, `reset_guest_vcpus()` and `reset_guest_mem()`?

File: lib/main_common.pm [hunk 1]
```diff
@@ -2729,7 +2729,12 @@ sub load_hypervisor_tests {
     }
 
     if (check_var('VIRT_PART', 'hotplugging')) {
-        loadtest 'virtualization/universal/hotplugging';    # Try to change properties of guests
```
Comment (alice-suse) [inline lib/main_common.pm:2732]: This hotplugging test is also used in developing product test, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L637. Please also update it .

File: lib/virt_autotest/utils.pm [hunk 2]
```diff
@@ -548,4 +549,92 @@ EOF
...
+
+    ## Network
+    # Remove temporary NIC devices, those are identified by a MAC starting with "$MAC_PREFIX"
+    remove_additional_nic($guest, $MAC_PREFIX);
+    ## Disk cleanup
+    my $disk_image = get_disk_image_name($guest);
+    remove_additional_disks($guest);
+    script_run("rm -f $disk_image");
+    ## CPU and memory
+    set_vcpus($guest, 2);
```
Comment (alice-suse) [inline lib/virt_autotest/utils.pm:617]: Should not it be reset to previous value before doing cpu hotplug test? BTW, this value is not fixed to 2 in developing product's test, and it is customized from outside.

**Comment by alice-suse** [comment]
@tbaev Can you also add verification run for developing product? Triggering two jobs against this PR in osd will be fine. KVM: http://openqa.suse.de/tests/7924359, XEN: http://openqa.suse.de/tests/7927277.

@waynechen55  should be interested in this and can also give a review.

**Comment by alice-suse** [comment]
@tbaev Can you also add verification run for developing product? Triggering two jobs against this PR in osd will be fine. KVM: http://openqa.suse.de/tests/7924359, XEN: http://openqa.suse.de/tests/7927277.

@waynechen55  should be interested in this and can also give a review.

**Comment by alice-suse** [comment]
BTW, there is  conflict in this PR when CI rebases, so need to manually rebase and then push code. Besides, better to merge the 3 commits into one.

**Comment by pdostal** [comment]
@tbaev OK, than I vote for `tests/virtualization/universal/hotplugging_utils.pm`.

**Comment by tbaev** [comment]
> @tbaev Can you also add verification run for developing product? Triggering two jobs against this PR in osd will be fine. KVM: http://openqa.suse.de/tests/7924359, XEN: http://openqa.suse.de/tests/7927277.
> 
> @waynechen55 should be interested in this and can also give a review.

@alice-suse 
The runs have failed after vCPU change on OSD.
[KVM](https://openqa.suse.de/tests/7940146#step/hotplugging_vCPUs/42)
[XEN](https://openqa.suse.de/tests/7940147#step/hotplugging_vCPUs/42)

@pdostal 
I agree that `tests/virtualization/universal/hotplugging_utils.pm` is a good solution.

File: schedule/virt_autotest/virt-guest-installation.yaml [hunk 3]
```diff
@@ -55,7 +55,12 @@ conditional_schedule:
...
         ENABLE_HOTPLUGGING:
             1:
-                - virtualization/universal/hotplugging
+                - virtualization/universal/hotplugging_guest_preparation
+                - virtualization/universal/hotplugging_network_interfaces
+                - virtualization/universal/hotplugging_HDD
+                - virtualization/universal/hotplugging_vCPUs
+                - virtualization/universal/hotplugging_memory
+                - virtualization/universal/hotplugging_cleanup
     storage_test:
```
Comment (Julie-CAO) [inline schedule/virt_autotest/virt-guest-installation.yaml:64]: Thank you for including the yaml file for TW tests.

**Comment by alice-suse** [comment]
@tbaev Do you close the PR by accident?

**Comment by tbaev** [comment]
> @tbaev Do you close the PR by accident?
Yes, my mistake I will make some changes and reopen it.

**Comment by pdostal** [comment]
@tbaev there is button for reopening. No need to create new one.
The new pull request is #13973.

**Pull request closed without merging**